apps/api/src/app/integrations/usecases/update-integration/update-integration.usecase.ts

```diff
@@ -29,6 +29,88 @@ export class UpdateIntegration {
     private disableNovuIntegration: DisableNovuIntegration
   ) {}
 
+  private async calculatePriorityAndPrimaryForActive({
+    existingIntegration,
+  }: {
+    existingIntegration: IntegrationEntity;
+  }) {
+    const result: { primary: boolean; priority: number } = {
+      primary: existingIntegration.primary,
+      priority: existingIntegration.priority,
+    };
+
+    const isChannelSupportsPrimary = CHANNELS_WITH_PRIMARY.includes(existingIntegration.channel);
+
+    const highestPriorityIntegration = await this.integrationRepository.findHighestPriorityIntegration({
+      _organizationId: existingIntegration._organizationId,
+      _environmentId: existingIntegration._environmentId,
+      channel: existingIntegration.channel,
+    });
+
+    if (highestPriorityIntegration?.primary) {
+      result.priority = highestPriorityIntegration.priority;
+      await this.integrationRepository.update(
+        {
+          _id: highestPriorityIntegration._id,
+          _organizationId: highestPriorityIntegration._organizationId,
+          _environmentId: highestPriorityIntegration._environmentId,
+        },
+        {
+          $set: {
+            priority: highestPriorityIntegration.priority + 1,
+          },
+        }
+      );
+    } else {
+      result.priority = highestPriorityIntegration ? highestPriorityIntegration.priority + 1 : 1;
+    }
+
+    const activeIntegrationsCount = await this.integrationRepository.countActiveExcludingNovu({
+      _organizationId: existingIntegration._organizationId,
+      _environmentId: existingIntegration._environmentId,
+      channel: existingIntegration.channel,
+    });
+    if (activeIntegrationsCount === 0 && isChannelSupportsPrimary) {
+      result.primary = true;
+    }
+
+    return result;
+  }
+
+  private async calculatePriorityAndPrimary({
+    existingIntegration,
+    active,
+  }: {
+    existingIntegration: IntegrationEntity;
+    active: boolean;
+  }) {
+    let result: { primary: boolean; priority: number } = {
+      primary: existingIntegration.primary,
+      priority: existingIntegration.priority,
+    };
+
+    if (active) {
+      result = await this.calculatePriorityAndPrimaryForActive({
+        existingIntegration,
+      });
+    } else {
+      await this.integrationRepository.recalculatePriorityForAllActive({
+        _id: existingIntegration._id,
+        _organizationId: existingIntegration._organizationId,
+        _environmentId: existingIntegration._environmentId,
+        channel: existingIntegration.channel,
+        exclude: true,
+      });
+
+      result = {
+        priority: 0,
+        primary: false,
+      };
+    }
```

Comment:
does this mean that if i update inactive integration it will update with a priority of 0 and primary false? 
if so it feels a bit like a side effect, would not we expect to get the same order we had before (if we deactivate and activate) so when we do recalculatePriorityForAllActive so the original order remains the same?

I hope I got it right. the side effect i think we have is that all the diactivated integration will go to the last priority right away but it could be an expected behavior

Reply:
> means the user doesn't want to use it.
or diactivated for some period of time due to internal issues or even the user is playing with the dashboard.
but at this stage as you said we can leave it like this and then update **if needed**